pennylane/ops/qubit/special_unitary.py

```diff
@@ -298,32 +434,240 @@ def compute_matrix(theta, num_wires):
             A = qml.math.tensordot(theta, pauli_basis_matrices(num_wires), axes=[[-1], [0]])
         return qml.math.expm(1j * A)
 
-    @staticmethod
-    def compute_decomposition(theta, wires, num_wires):
-        r"""Representation of the operator as a product of other operators (static method).
+    def get_one_parameter_generators(self, interface=None):
+        r"""Compute the generators of one-parameter groups that reproduce
+        the partial derivatives of a special unitary gate.
 
-        .. math:: O = O_1 O_2 \dots O_n.
+        Args:
+            interface (str): The auto-differentiation framework to be used for the
+                computation.
 
-        This ``Operation`` is decomposed into the corresponding ``QubitUnitary``.
+        Returns:
+            tensor_like: The generators for one-parameter groups that reproduce the
+            partial derivatives of the special unitary gate.
+            There are :math:`d=4^n-1` generators for :math:`n` qubits, so that the
+            output shape is ``(4**num_wires-1, 2**num_wires, 2**num_wires)``.
+
+        Consider a special unitary gate parametrized in the following way:
+
+        .. math::
+
+            U(\theta) &= \exp\{A(\theta)\}\\
+            A(\theta) &= \sum_{m=1}^d i \theta_m P_m\\
+            P_m &\in \{I, X, Y, Z\}^{\otimes n} \setminus \{I^{\otimes n}\}
+
+        Then the partial derivatives of the gate can be shown to be given by
+
+        .. math::
+
+            \frac{\partial}{\partial \theta_\ell} U(\theta) &= U(\theta)
+            \frac{\mathrm{d}}{\mathrm{d}t}\exp\left(t\Omega_\ell(\theta)\right)\large|_{t=0}
+
+        where :math:`\Omega_\ell(\theta)` is the one-parameter generator belonging to the partial
+        derivative :math:`\partial_\ell U(\theta)` at the parameters :math:`\theta`.
+        It can be computed via
+
+        .. math::
+
+            \Omega_\ell(\theta) &= U(\theta)^\dagger
+            \left(\frac{\partial}{\partial \theta_\ell}\mathfrak{Re}[U(\theta)]
+            +i\frac{\partial}{\partial \theta_\ell}\mathfrak{Im}[U(\theta)]\right)
+
+        where we may compute the derivatives :math:`\frac{\partial}{\partial \theta_\ell} U(\theta)` using auto-differentiation.
 
-        .. seealso:: :meth:`~.SpecialUnitary.decomposition`.
+        .. warning::
+
+            An auto-differentiation framework is required for this function.
+            The matrix exponential is not differentiable in Autograd. Therefore this function
+            only supports JAX, Torch and Tensorflow.
+
+        ..seealso:: `~.SpecialUnitary`
+
+        """
+        theta = self.data[0]
+        if len(qml.math.shape(theta)) > 1:
```

Comment:
You're right. I'll revisit what was the reason here.

Reply:
Let's open an issue for broadcasting and differentiation :+1:

Reply:
#3721

Reply:
Wait, so JAX's `expm` does now support broadcasting? Then what problem do we have? Why is broadcasting not compatible with hardware-based gradient transforms?

Reply:
Yes and no. Jax's `expm` does not support broadcasting, I manually dispatch to iterating over the broadcasting axis when using JAX, see [here](https://github.com/PennyLaneAI/pennylane/blob/58d8243f414e52b5e344d60a81a12897576caceb/pennylane/ops/qubit/special_unitary.py#L307)

For the broadcasting-compatible differentiation I see some additional work coming that probably will also require benchmarking. This is because differentiation typically does not support broadcasting, i.e. a function with shape signature
```python
(batch_dim, *input_dims) -> (batch_dim, *output_dims)
```
will have a Jacobian of shape `(batch_dim, *output_dims, batch_dim, *input_dims)` but we are only interested in the diagonal across the two batch dimensions.
The question then is, can we afford to compute the too large Jacobian, or do we want to go for manual stacking of the non-broadcasted Jacobians?